src/localizePropsDecorator.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { skinnable, contains } from 'revenge';
+import { t, props } from 'tcomb-react';
+import mapKeys from 'lodash/mapKeys';
+import mapValues from 'lodash/mapValues';
+import displayName from './displayName';
+
+export default function localizePropsDecorator({ containerNamespace, local }) {
+  const globalizeLocalKeys = obj => mapKeys(obj, (_, k) => {
+    if (local[k]) {
+      return `${containerNamespace}${k}`;
+    }
+    return k;
+  });
+  const localizeLocalKeys = obj => mapKeys(obj, (_, k) => {
+    return k.replace(containerNamespace, '');
+  });
+  const globalizedLocalTypes = globalizeLocalKeys(local || {});
+  const localizeProps = props => localizeLocalKeys({
+    ...props,
+    transition: (...args) => {
+      if (args.length === 1 && t.Object.is(args[args.length - 1])) {
+        return props.transition(globalizeLocalKeys(args[args.length - 1]));
+      }
+      throw new Error(`Sorry, local transitions do not yet support arguments ${args.map(v => typeof v).join(',')}`);
```

Comment:
in other words, we don't support the forms `transition(state => state)` or `transition(id, patchOrFn)` that `state` internally supports.

This is not an issue since both are not used typically + it is basically impossible to support the "function" form from here (while it will be easy if and when the "local state" concept is at `state` level)